pyke/lightcurve.py

```diff
@@ -93,37 +93,36 @@ def flatten(self, window_length=101, polyorder=3, **kwargs):
     def fold(self, phase, period):
         return LightCurve(((self.time - phase + 0.5 * period) / period) % 1 - 0.5, self.flux)
 
-    def draw(self):
-        raise NotImplementedError("Should we implement a LightCurveDrawer class?")
-
     def to_csv(self):
         raise NotImplementedError()
 
-    def plot(self, ax=None, norm=True,**kwargs):
+    def plot(self, t=None, flux=None, flux_err=None, ax=None, norm=True, xlabel = 'Time - 2454833 (days)',
+            ylabel = 'Normalised Flux',color='#363636', fill=True, grid=True, **kwargs):
         """
         Plot a Light Curve.
         """
         if ax is None:
             fig, ax = plt.subplots(1)
-        t0=float(int(self.time[0] / 100) * 100.0)
-        f = self.flux
-        ferr = self.flux_err
+        if t == None:
+            t = self.time
+        if flux == None:
+            flux = self.flux
+        if flux_err == None:
+            flux_err = self.flux_err
```

Comment:
Maybe is not necessary to have ``t=None, flux=None, flux_err=None`` as arguments, since they are already attributes in the class? I would just use ``self.time, self.flux,`` and `` self.flux_err`` directly.

Reply:
I did this on purpose incase we wanted to specify e.g. folded time

Reply:
Because this is a class method, it would be poor object-oriented design to accept `time`, `flux`, and `flux_err` because they are class attributes.  Instead, the preferred way to plot a lightcurve with custom time or flux should be `LightCurve(time, flux).plot()`.  A folded lightcurve can be plotted using `lc.fold(...).plot()`.

Reply:
Agreed with @barentsen